src/main/scala/com/gu/fastly/Lambda.scala

```diff
@@ -37,6 +46,18 @@ class Lambda {
       }
     }
 
+    val recordsAddedResult: Seq[UserRecordResult] = Await.result(
```

Comment:
I think the CrierEventProcessor.process code above might be asynchronous, so there's no guarantee that this would happen after the fastly purge request is sent? So might be worth calling from within the `match` - that way you can also throw away updates that aren't of type `Content` or `Delete`

Reply:
also might be worth pulling all of this logic out into a separate function which can then be wrapped in a `Try` and maybe some timeout stuff so we can prevent the pushing messages onto kinesis bit of code from interfering with the other cache purging tasks